.github/workflows/precommit.yml

```diff
@@ -302,7 +330,8 @@ jobs:
 
   test-snowpark-pandas:
     name: Test modin-${{ matrix.os }}-${{ matrix.python-version }}-${{ matrix.cloud-provider }}
-    needs: build
+    needs: build, changes
+    if: ${{ needs.changes.outputs.snowpark_pandas == 'true' }}
```

Comment:
snowpark pandas tests are suppose to be run for both snowpark and snowpark pandas changes, because any snowpark changes can impact snowpark pandas functionality